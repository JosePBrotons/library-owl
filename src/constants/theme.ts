import { Dimensions } from 'react-native';
const { width, height } = Dimensions.get('window');

export const COLORS = {
    black: 'rgba(30, 31, 32, 1)',
    white: 'rgba(255, 255, 255, 1)',
    whiteTwo: 'rgba(248, 248, 248, 1)',
    gray: 'rgba(106, 106, 106, 1)',
};
export const SIZES = {
    base: 8,
    font: 14,
    radius: 12,
    padding: 24,

    h1: 30,
    h2: 22,
    h3: 16,
    h4: 14,
    body1: 30,
    body2: 22,
    body3: 16,
    body4: 14,

    width,
    height,
};
export const FONTS = {
    h1: { fontFamily: 'Nunito-Black', fontSize: SIZES.h1, lineHeight: 36 },
    h2: { fontFamily: 'Nunito-Bold', fontSize: SIZES.h2, lineHeight: 30 },
    h3: { fontFamily: 'Nunito-Bold', fontSize: SIZES.h3, lineHeight: 22 },
    h4: { fontFamily: 'Nunito-Bold', fontSize: SIZES.h4, lineHeight: 22 },
    body1: {
        fontFamily: 'Nunito-Regular',
        fontSize: SIZES.body1,
        lineHeight: 36,
    },
    body2: {
        fontFamily: 'Nunito-Regular',
        fontSize: SIZES.body2,
        lineHeight: 30,
    },
    body3: {
        fontFamily: 'Nunito-Regular',
        fontSize: SIZES.body3,
        lineHeight: 22,
    },
    body4: {
        fontFamily: 'Nunito-Regular',
        fontSize: SIZES.body4,
        lineHeight: 22,
    },
};

const appTheme = { COLORS, SIZES, FONTS };

export default appTheme;
