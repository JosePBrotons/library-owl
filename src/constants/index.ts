import theme, { COLORS } from './theme';

export { theme, COLORS };
