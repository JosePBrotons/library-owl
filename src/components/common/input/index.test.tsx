import React from 'react';
import renderer from 'react-test-renderer';
import Input from '.';
import { COLORS } from './../../../constants';

test('renders input component correctly', () => {
    const props = {
        placeholder: 'Email',
        value: 'me@josepbrotons.com',
        placeholderTextColor: COLORS.lightGray,
        onChangeText: jest.fn,
        iconName: 'mail-reply',
    };
    const inputCmp = renderer.create(<Input {...props} />).toJSON();
    expect(inputCmp).toMatchSnapshot();
});
