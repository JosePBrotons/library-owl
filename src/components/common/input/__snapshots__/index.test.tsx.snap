// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders input component correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "rgba(248, 248, 248, 1)",
      "borderRadius": 5,
      "color": "rgba(69, 69, 69, 1)",
      "flexDirection": "row",
      "height": 40,
      "marginBottom": 20,
      "paddingHorizontal": 10,
      "width": "80%",
    }
  }
>
  <Text
    allowFontScaling={false}
    style={
      Array [
        Object {
          "color": "rgba(233, 233, 233, 1)",
          "fontSize": 16,
        },
        undefined,
        Object {
          "fontFamily": "FontAwesome",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
        Object {},
      ]
    }
  >
    ï„’
  </Text>
  <TextInput
    allowFontScaling={true}
    onChangeText={[Function]}
    placeholder="Email"
    placeholderTextColor="rgba(233, 233, 233, 1)"
    rejectResponderTermination={true}
    secureTextEntry={false}
    style={
      Object {
        "color": "rgba(69, 69, 69, 1)",
        "flex": 1,
        "fontFamily": "Nunito-Regular",
        "fontSize": 14,
        "lineHeight": 22,
        "paddingLeft": 8,
      }
    }
    underlineColorAndroid="transparent"
    value="me@josepbrotons.com"
  />
</View>
`;
